File: leetcode/easy/intersectionOfTwoArrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
        unordered_set<int> s;
        for(int i : nums1) s.insert(i); //store all of nums1 in a set
        for(int i : nums2) {
            if(s.find(i) != s.end()){ //if i in nums2 is in nums1
                res.push_back(i); //push_back i
                s.erase(s.find(i)); //remove it from the set
            }
        }
        return res;
    }
};

File: leetcode/easy/intersectionOfTwoArrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res = []
        s = {i for i in nums1} #store nums1 in a set
        #a set does not store duplicate values, so we only get one occurance of each i in nums1          in s
        for i in nums2:
            if i in s:
                res.append(i) #append it to res
                s.remove(i) # remove is from s so you only have unique values
        return res
        
